dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -1,36 +1,49 @@
 package org.dataland.datalandbackend.services
 
 import com.fasterxml.jackson.databind.ObjectMapper
+import org.dataland.datalandbackend.entities.DataMetaInformationEntity
 import org.dataland.datalandbackend.model.DataType
 import org.dataland.datalandbackend.model.StorableDataSet
+import org.dataland.datalandbackend.model.enums.data.QAStatus
 import org.dataland.datalandbackendutils.exceptions.InternalServerErrorApiException
 import org.dataland.datalandbackendutils.exceptions.InvalidInputApiException
 import org.dataland.datalandbackendutils.exceptions.ResourceNotFoundApiException
 import org.dataland.datalandinternalstorage.openApiClient.api.StorageControllerApi
 import org.dataland.datalandinternalstorage.openApiClient.infrastructure.ServerException
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
 import org.slf4j.LoggerFactory
+import org.springframework.amqp.core.Message
+import org.springframework.amqp.rabbit.annotation.RabbitListener
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.context.annotation.ComponentScan
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 import org.springframework.transaction.annotation.Transactional
+import java.util.*
 
 /**
  * Implementation of a data manager for Dataland including metadata storages
  * @param objectMapper object mapper used for converting data classes to strings and vice versa
  * @param companyManager service for managing company data
  * @param metaDataManager service for managing metadata
  * @param storageClient service for managing data
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
 */
-@ComponentScan(basePackages = ["org.dataland"])
 @Component("DataManager")
 class DataManager(
     @Autowired var objectMapper: ObjectMapper,
     @Autowired var companyManager: CompanyManager,
     @Autowired var metaDataManager: DataMetaInformationManager,
     @Autowired var storageClient: StorageControllerApi,
-
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
 ) {
+    @Value("\${spring.rabbitmq.upload-queue}")
+    private val uploadQueue = ""
+
+    @Value("\${spring.rabbitmq.storage-queue}")
+    private var storageQueue = ""
```

Comment:
why is one a val, and one a var?

Reply:
Excellent question. This was not supposed to stay. I just wanted to see the IDE's reaction using the annotation on different field types. It's fixed now

Reply:
ok